cpp/scenario/gazebo/src/Joint.cpp

```diff
@@ -504,9 +504,20 @@ bool Joint::setAccelerationTarget(const double acceleration, const size_t dof)
 
 bool Joint::setGeneralizedForceTarget(const double force, const size_t dof)
 {
-    if (this->controlMode() != base::JointControlMode::Force) {
-        gymppError << "The active joint control mode does not accept a force"
-                   << std::endl;
+    const std::vector<base::JointControlMode> allowedControlModes = {
+        base::JointControlMode::Force,
+        base::JointControlMode::Position,
+        base::JointControlMode::PositionInterpolated,
+        base::JointControlMode::Velocity};
+
+    auto it = std::find(allowedControlModes.begin(),
+                        allowedControlModes.end(),
+                        this->controlMode());
+
+    if (it == allowedControlModes.end()) {
+        gymppError
+            << "The active joint control mode does not accept a force target"
+            << std::endl;
```

Comment:
I am not sure I follow. Why it is legal to call `setGeneralizedForceTarget` on a Position-controlled joint?

Reply:
Let's consider a position controlled model (either with the default JointController system with an independent PID for each joint or a custom controller like the ComputedTorque fixed base). The user will call the `Model::setJointPositionTargets`, but the controller running as a system will use the pointer to the same model to call `Model::setJointGeneralizedForceTargets`.

Do you see any other possibility?

Reply:
Ok, that make sense, but that it will be possible with any control mode you can add in the future, so in that case I don't understand why there is the need for the `allowedControlModes` check.

Reply:
The default control mode of a joint is `Idle`:

https://github.com/robotology/gym-ignition/blob/b1ff4419743c71e49e457b42213170a6c0f2f415/cpp/scenario/gazebo/src/Joint.cpp#L140-L142

Reply:
Ah, I  see. Just to understand, so you want to make sure that every future control mode to fail  to set torque, right? Otherwise it would be sufficient to just check if the control mode is not Idle.

Reply:
Even though I don't think we'll add anytime soon a new control mode, but yes, that's the reason.